esi_ui/api/esi_api.py

```diff
@@ -59,10 +61,42 @@ def node_list(request):
             'resource_class': lease.resource_class,
             'start_time': lease.start_time,
             'end_time': lease.end_time,
-            'status': lease.status
+            'status': lease.status,
+            'lease_duration': [lease.start_time.replace('T', ' ', 1) + " - " + lease.end_time.replace('T', ' ', 1)],
+            'leases': [{ 'id': lease.id, 'duration': lease.start_time.replace('T', ' ', 1) + " - " + lease.end_time.replace('T', ' ', 1), 'active': lease.status == 'active', 'displayText': lease.id + ' (' + lease.start_time.replace('T', ' ', 1) + ' - ' + lease.end_time.replace('T', ' ', 1) + ')' }]
         }
         for lease, e in lease_node_list
     ]
+    
+    for node in node_infos:
```

Comment:
Instead of having a double for loop after a for loop, I would have a node-to-list-of-leases map created with each list of leases being sorted by start time. Then instead of having to double for loop to update and remove nodes, just use the map for one pass through.